src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc

```diff
@@ -0,0 +1,238 @@
+using test::components::application_manager_test::MockHMICapabilities;
+using test::components::application_manager_test::MockCommandFactory;
+using test::components::policy_test::MockPolicyHandlerInterface;
+
+namespace rc_rpc_plugin_test {
+using namespace rc_rpc_plugin;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kAppId = 5u;
+const int kModuleId = 153u;
+const std::string kResource = "CLIMATE";
+}  // namespace
+
+class RCGetInteriorVehicleDataConsentTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  RCGetInteriorVehicleDataConsentTest()
+      : mock_app_(utils::MakeShared<NiceMock<MockApplication> >())
+      , command_holder(app_mngr_)
+      , request_controller(mock_request_controler)
+      , rpc_service(app_mngr_,
+                    request_controller,
+                    &mock_protocol_handler,
+                    &mock_hmi_handler,
+                    command_holder)
+      , rc_app_extention_(utils::MakeShared<RCAppExtension>(kModuleId))
+      , mock_rpc_plugin_manager(
+            utils::MakeShared<NiceMock<MockRPCPluginManager> >()) {
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kAppId));
+    ON_CALL(app_mngr_, hmi_interfaces())
+        .WillByDefault(ReturnRef(mock_hmi_interfaces_));
+    ON_CALL(
+        mock_hmi_interfaces_,
+        GetInterfaceState(application_manager::HmiInterfaces::HMI_INTERFACE_RC))
+        .WillByDefault(Return(application_manager::HmiInterfaces::
+                                  InterfaceState::STATE_AVAILABLE));
+    ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app_));
+    ON_CALL(mock_allocation_manager_, GetApplicationExtention(_))
+        .WillByDefault(Return(rc_app_extention_));
+    ON_CALL(app_mngr_, GetPolicyHandler())
+        .WillByDefault(ReturnRef(mock_policy_handler_));
+    ON_CALL(app_mngr_, hmi_capabilities())
+        .WillByDefault(ReturnRef(mock_hmi_capabilities_));
+    ON_CALL(mock_hmi_capabilities_, rc_capability())
+        .WillByDefault(Return(&rc_capabilities_));
+    ON_CALL(mock_policy_handler_,
+            CheckHMIType(
+                _, mobile_apis::AppHMIType::eType::REMOTE_CONTROL, nullptr))
+        .WillByDefault(Return(true));
+    ON_CALL(mock_allocation_manager_, is_rc_enabled())
+        .WillByDefault(Return(true));
+  }
+
+  template <class Command>
+  application_manager::SharedPtr<Command> CreateRCCommand(
+      MessageSharedPtr& msg) {
+    InitCommand(kDefaultTimeout_);
+    return ::utils::MakeShared<Command>(msg ? msg : msg = CreateMessage(),
+                                        app_mngr_,
+                                        rpc_service,
+                                        mock_hmi_capabilities_,
+                                        mock_policy_handler_,
+                                        mock_allocation_manager_);
+  }
+
+  MessageSharedPtr CreateBasicMessage() {
```

Comment:
@ValeriiMalkov add comments. What avtualy basic message mean? What the dufferent between CreateBasicMessage and  CreateMessage ?